src/main/kotlin/com/defenseunicorns/koci/Router.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2025 Defense Unicorns
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.defenseunicorns.koci
+
+import io.ktor.http.*
+import java.net.URI
+
+private fun URLBuilder.paginate(n: Int, last: String? = null): URLBuilder = apply {
+    parameters.append("n", n.toString())
+    last?.let { parameters.append("last", it) }
+}
+
+class Router(registryURL: String) {
+    private val v2Prefix = "v2/"
+
+    private val base: URLBuilder = URLBuilder().takeFrom(registryURL).appendPathSegments(v2Prefix)
+
+    fun base(): Url {
+        return base.build()
+    }
+
+    fun catalog(): Url {
+        return base.clone().appendPathSegments("_catalog").build()
+    }
+
+    fun catalog(n: Int, lastRepo: String? = null): Url {
+        val catalogBase = URLBuilder().takeFrom(catalog())
```

Comment:
Idk what the "kotlin" way to do this is but why does this redeclare the base vs base.close() like the other ones?

Reply:
simplified through slightly less dry code, but the locality of behavior is within 2 lines